tests/client/ayon_core/pipeline/editorial/test_media_range_with_retimes.py

```diff
@@ -0,0 +1,149 @@
+    expected_retimed_data: dict,
+    handle_start: int = 10,
+    handle_end: int = 10,
+):
+    file_path = os.path.join(_RESOURCE_DIR, file_name)
+    otio_clip = otio.schema.Clip.from_json_file(file_path)
+
+    retimed_data = get_media_range_with_retimes(
+        otio_clip, handle_start, handle_end
+    )
+    assert retimed_data == expected_retimed_data
+
+
+def test_movie_with_end_handle_end_only():
+    """
+    Movie clip (no embedded timecode)
+    available_range = 0-171 25fps
+    source_range = 0-16450 25fps
+    """
+    expected_data = {
+        'mediaIn': 0.0,
+        'mediaOut': 170.0,
+        'handleStart': 0,
+        'handleEnd': 10,
+        'speed': 1.0
+    }
+    _check_expected_retimed_values(
+        "movie_with_handles.json",
+        expected_data,
+    )
+
+
+def test_movie_embedded_tc_handle():
+    """
+    Movie clip (embedded timecode 1h)
+    available_range = 86400-86500 24fps
+    source_range = 90032-90076 25fps
+    """
+    expected_data = {
+        'mediaIn': 30.720000000001164,
+        'mediaOut': 71.9600000000064,
+        'handleStart': 10,
+        'handleEnd': 10,
+        'speed': 1.0
+    }
+    _check_expected_retimed_values(
+        "qt_embedded_tc.json",
+        expected_data
+    )
+
+
+def test_movie_retime_effect():
+    """
+    Movie clip (embedded timecode 1h)
+    available_range = 0-171 25fps
+    source_range = 0-16450 25fps
+    retimed speed: 250%
+    """
+    expected_data = {
+        'mediaIn': 0.0,
+        'mediaOut': 426.5,
+        'handleStart': 0,
+        'handleEnd': 25,
+        'speed': 2.5,
+        'versionData': {
+            'retime': True,
+            'speed': 2.5,
+            'timewarps': [],
+            'handleStart': 0,
+            'handleEnd': 25
+        }
+    }
+#    import rpdb ; rpdb.Rpdb().set_trace()
+    _check_expected_retimed_values(
+        "qt_retimed_speed.json",
+        expected_data
+    )
+
+
+def test_img_sequence_no_handles():
+    """
+    Img sequence clip (no embedded timecode)
+    available files = 1000-1100
+    source_range =  0-100 25fps
+    """
+    expected_data = {
+        'mediaIn': 1000,
+        'mediaOut': 1100,
+        'handleStart': 0,
+        'handleEnd': 0,
+        'speed': 1.0
+    }
+    _check_expected_retimed_values(
+        "img_seq_no_handles.json",
+        expected_data
+    )
+
+
+def test_img_sequence_with_handles():
+    """
+    Img sequence clip (no embedded timecode)
+    available files = 1000-1100
+    source_range =  34-72 25fps
+    """
+    expected_data = {
+        'mediaIn': 1034,
+        'mediaOut': 1072,
+        'handleStart': 10,
+        'handleEnd': 10,
+        'speed': 1.0
+    }
+    _check_expected_retimed_values(
+        "img_seq_with_handles.json",
+        expected_data
+    )
+
+
+def test_img_sequence_with_embedded_tc_and_handles():
+    """
+    Img sequence clip (embedded timecode 1h)
+    available files = 1000-1100
+    source_range =  91046.625-91,120.625 25fps
```

Comment:
This value hint is unclear to me, could you elaborate a bit on what does it refer to?

Reply:
Totally, so embedded timecode for the img sequence is 1h meaning 90000 frames at 25fps.
Absolute soure range being `91046.625 - 91120.625` that gives us relative source range `1046.625 - 1120.625`.

So in exprimed as seconds the img sequence clip source range is:
- start_time : `1046 frames / 25 fps` = `41,84` seconds
- duration : `1120.625 - 1046.625` = `74 frames / 25 fps` = `2.94` seconds

Now as recompute duration as frames with an available range at 24.0 fps:
- start_time = `41,84` seconds x `24.0` fps = `1004.16` frames
- duration = `2.94` seconds x `24.0fps` = `70.56` frames

Finally remapping those as discrete frame numbers to the sequence:
![Untitled-2024-09-19-1305](https://github.com/user-attachments/assets/35c29575-5ee7-4939-b0bc-291a370c5795)

- first frame displayed is frame at duration `1004.16` = frame number `1005`
- last frame displayed is frame at duration `1004.16 + 70.56 = 1074.72` = frame number `1075`

Hopes that clarifies it.

Reply:
I see, so it seems to be just a typo: `91046.625-91,120.625 25fps` should be actually `91046.625-91120.625 25fps`. The inconsistency of coma had confused me. Thank you.

Reply:
> first frame displayed is frame at duration 1004.16 = frame number 1005

The calculation makes sense for sure the only point we need to validate is if the rounding up even below **.5** will work.

Reply:
Oh missed that, thanks for this ! It's adjusted.
For the rounding work below .5, do you want another tests for that ? 
Confirm this is properly handled by otio:
![image](https://github.com/user-attachments/assets/d589c957-79c0-4397-ab09-b900aba19922)